Hogwarts/src/test/java/com/example/Hogwarts/service/HogwartsFacultyApplicationMockTests.java

```diff
@@ -0,0 +1,93 @@
+package com.example.Hogwarts.service;
+import com.example.Hogwarts.controller.FacultyController;
+import com.example.Hogwarts.model.Faculty;
+import com.example.Hogwarts.model.Student;
+import com.example.Hogwarts.repository.FacultyRepository;
+import com.example.Hogwarts.repository.StudentRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import java.util.Optional;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@WebMvcTest(FacultyController.class)
+public class HogwartsFacultyApplicationMockTests {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @SpyBean
+    private FacultyService facultyService;
```

Comment:
Шпионить над интерфейсом мы не можем
```suggestion
    private FacultyServiceImpl facultyService;
```